include/lbann/data_store/generic_data_store.hpp

```diff
@@ -82,30 +79,40 @@ class generic_data_store {
     //m_cur_idx = 0;
   }
 
-  virtual void get_data_buf(std::string dir, std::string filename, std::vector<unsigned char> *&buf, int tid) = 0;
-
-  virtual void get_data_buf(int data_id, std::vector<unsigned char> *&buf, int tid) = 0; 
+  /// image data readers call this method
+  /// @todo: do we need tid?
```

Comment:
You can always just run `omp_get_thread_num` instead of passing the `tid` explicitly in the cases it's needed.

Reply:
In the existing data_store_imagenet and data_store_multi_images the tid is not needed. I sort of doubt it will be needed in other data_stores, so, per Nikoli's suggestion, I'll remove it.

Reply:
sounds good.